Libraries/Opc.Ua.Client/Subscription/Subscription.cs

```diff
@@ -2053,6 +2048,21 @@ uint revisedLifetimeCounter
             }
         }
 
+        /// <summary>
+        /// Calculate the KeepAliveInterval based on <see cref="m_currentPublishingInterval"/> and <see cref="m_currentKeepAliveCount"/>
+        /// </summary>
+        /// <returns></returns>
+        private int CalculateKeepAliveInterval()
+        {
+            int keepAliveInterval = (int)(Math.Min(m_currentPublishingInterval * (m_currentKeepAliveCount + 1), Int32.MaxValue));
+            if (m_keepAliveInterval < kMinKeepAliveTimerInterval)
```

Comment:
bug, use keepAliveInterval
```suggestion
            if (keepAliveInterval < kMinKeepAliveTimerInterval)
```